develop/api_guc.rst

```diff
@@ -222,6 +222,19 @@ The default value for this GUC is 'on'.
 Planner Configuration
 ------------------------------------------------
 
+citus.local_table_join_policy (enum)
+$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+
+Sets how citus should plan local and distributed table joins. There are four modes available:
```

Comment:
should we very briefly mention why would anyone need to change this? Also, maybe move `auto` to top as that's the default?